snippets/paths.py

```diff
@@ -41,7 +43,7 @@
     account=wallet.address,
     amount=destination_amount,
     destination=destination_account,
-    paths=paths,
+    paths=[[PathStep(**path_data) for path_data in path] for path in paths],
```

Comment:
Due to the more rigorous type checks, developers cannot "round-trip" the values i.e. the output of ripple-path-find is not longer compatible with the `Payment` transaction.

Can we write a utility (explicit) type-cast constructor that takes an input JSON and returns a 2-dimensional nested `PathStep` array?

```
def convertPathsJsonToList(paths: Dict[ ??? ]) -> List[List[PathStep]]:
    return [[PathStep(**path_data) for path_data in path] for path in paths]
```

Reply:
I thought dictionaries weren't handled properly in the `from_xrpl` logic, but turns out they are, so I added an extra check to allow dictionaries in place of models.